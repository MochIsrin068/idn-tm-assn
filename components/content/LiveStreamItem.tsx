import React from "react";
import { Typography } from "antd";
import { EyeOutlined } from "@ant-design/icons";

const { Text, Title } = Typography;

type TPropsLiveStreamItem = {
  data: {
    title: string;
    date: string;
    time: string;
    category: string;
    thumbnail: string;
    status: string;
    views: string;
  };
};

export default function LiveStreamItem({ data }: TPropsLiveStreamItem) {
  const isScheduler = data.status.toLowerCase() === "terjadwal";
  const isLive = data.status.toLowerCase() === "live";

  return (
    <div
      className="rounded-lg flex flex-col justify-between h-[25vh] hover:opacity-90"
      style={{
        background: `url("${data.thumbnail}")`,
        backgroundSize: "cover",
      }}
      data-testid="live-stream-container"
    >
      {isLive ? (
        <div className="mx-3 my-3 p-1 flex items-center bg-[#00000047] w-36 justify-center rounded-lg">
          <EyeOutlined className="text-white text-sm" />
          <Text className="text-white text-sm">{`${data.views} |`}</Text>

          <div className="bg-red-500 h-2 w-2 mx-1 rounded-lg" />
          <Text className="text-red-500 text-sm">{data.category}</Text>
        </div>
      ) : (
        <div
          className={`m-4 px-2 py-1 ${
            isScheduler ? "bg-red-200" : "bg-gray-400"
          } ${
            isScheduler ? "text-red-500" : "text-white"
          }  w-[78px] rounded-lg text-sm flex items-center justify-center`}
        >
          {data.status}
        </div>
      )}

      <div className="p-4 bg-[#0000000d]">
        <Title level={5} className="!m-0 !text-white line-clamp-2">
          {data.title}
        </Title>
        <Text className="text-white">{data.category}</Text>
      </div>
    </div>
  );
}
