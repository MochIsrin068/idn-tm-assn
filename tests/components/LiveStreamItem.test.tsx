import React from "react";
import { render, screen } from "@testing-library/react";
import LiveStreamItem from "../../components/content/LiveStreamItem";

const mockProps = {
  data: {
    title: "Nyanyi-nyanyi malem enak nih",
    date: "2022-08-25",
    time: "09.10",
    category: "Music",
    thumbnail: "https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg",
    status: "Live",
    views: "10,5rb",
  },
};

describe("tests components LiveStreamItem", () => {
  it("Perform snapshots test LiveStreamItem", () => {
    const inputDefaultTree = render(<LiveStreamItem {...mockProps} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test background container use thambnail data", () => {
    render(<LiveStreamItem {...mockProps} />);
    const container = screen.getByTestId("live-stream-container");
    expect(container).toHaveStyle(
      `background: url(https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg)`
    );
  });

  it("test text views if isView rendered on screen", () => {
    render(<LiveStreamItem {...mockProps} />);
    const text = screen.getByText(/10,5rb/i);
    expect(text).toBeInTheDocument();
  });

  it("test text status if isView === false rendered on screen", () => {
    render(
      <LiveStreamItem data={{ ...mockProps.data, status: "terjadwal" }} />
    );
    const text = screen.getByText(/terjadwal/i);
    expect(text).toBeInTheDocument();
  });

  it("test text title rendered on screen", () => {
    render(<LiveStreamItem {...mockProps} />);
    const text = screen.getByText(/Nyanyi-nyanyi malem enak nih/i);
    expect(text).toBeInTheDocument();
  });

  it("test text category rendered on screen", () => {
    render(
      <LiveStreamItem data={{ ...mockProps.data, status: "terjadwal" }} />
    );
    const text = screen.getByText(/Music/i);
    expect(text).toBeInTheDocument();
  });

  // optional checker and need query by text becouse that not rendered
  it("test text views wrong if isView rendered on screen", () => {
    render(<LiveStreamItem {...mockProps} />);
    const text = screen.queryByText(/10,5rb test/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text status wrong if isView === false rendered on screen", () => {
    render(
      <LiveStreamItem data={{ ...mockProps.data, status: "terjadwal" }} />
    );
    const text = screen.queryByText(/terjadwal test/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text title wrong rendered on screen", () => {
    render(<LiveStreamItem {...mockProps} />);
    const text = screen.queryByText(/Nyanyi-nyanyi malem enak nih test/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text category wrong rendered on screen", () => {
    render(<LiveStreamItem {...mockProps} />);
    const text = screen.queryByText(/Music test/i);
    expect(text).not.toBeInTheDocument();
  });
});
