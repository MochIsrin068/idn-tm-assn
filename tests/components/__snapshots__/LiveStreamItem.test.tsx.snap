// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components LiveStreamItem Perform snapshots test LiveStreamItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-lg flex flex-col justify-between h-[25vh] hover:opacity-90"
        data-testid="live-stream-container"
        style="background: url(https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg); background-size: cover;"
      >
        <div
          class="mx-3 my-3 p-1 flex items-center bg-[#00000047] w-36 justify-center rounded-lg"
        >
          <span
            aria-label="eye"
            class="anticon anticon-eye text-white text-sm"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="eye"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
              />
            </svg>
          </span>
          <span
            class="ant-typography text-white text-sm"
          >
            10,5rb |
          </span>
          <div
            class="bg-red-500 h-2 w-2 mx-1 rounded-lg"
          />
          <span
            class="ant-typography text-red-500 text-sm"
          >
            Music
          </span>
        </div>
        <div
          class="p-4 bg-[#0000000d]"
        >
          <h5
            class="ant-typography !m-0 !text-white line-clamp-2"
          >
            Nyanyi-nyanyi malem enak nih
          </h5>
          <span
            class="ant-typography text-white"
          >
            Music
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-lg flex flex-col justify-between h-[25vh] hover:opacity-90"
      data-testid="live-stream-container"
      style="background: url(https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg); background-size: cover;"
    >
      <div
        class="mx-3 my-3 p-1 flex items-center bg-[#00000047] w-36 justify-center rounded-lg"
      >
        <span
          aria-label="eye"
          class="anticon anticon-eye text-white text-sm"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="eye"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
        </span>
        <span
          class="ant-typography text-white text-sm"
        >
          10,5rb |
        </span>
        <div
          class="bg-red-500 h-2 w-2 mx-1 rounded-lg"
        />
        <span
          class="ant-typography text-red-500 text-sm"
        >
          Music
        </span>
      </div>
      <div
        class="p-4 bg-[#0000000d]"
      >
        <h5
          class="ant-typography !m-0 !text-white line-clamp-2"
        >
          Nyanyi-nyanyi malem enak nih
        </h5>
        <span
          class="ant-typography text-white"
        >
          Music
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
